### ISSUE TITLE
Оптимизация небольшого количества пуль

### ISSUE THREAD

**Issue by CREAsTIVE** [vectozavr/shooter#12]
Даже 20 выстреленных пуль одновременно сильно тормозят, предположительно проблема в рейкастинге

**Comment by vectozavr** [comment]
Да, проблема в рейкастинге: сейчас для обнаружения пересечений мы проходим циклом по всем треугольникам. Я думаю ввести проверку на близость луча с обьектом чтобы отбрасывать при поиске все далекие объекты.

**Comment by CREAsTIVE** [comment]
А с физикой получается такая же проблема?

**Comment by vectozavr** [comment]
С физикой там тоже есть узкие места: мы проверяем коллизию всех объектов со всеми (сложность N**2). Можно либо добавить проверку на дальность объектов, либо разбивать пространство на блоки.
